airflow/utils/helpers.py

```diff
@@ -147,6 +148,68 @@ def chain(*tasks):
         up_task.set_downstream(down_task)
 
 
+def popen_and_tail(cmd, shell=False, bufsize=512, env=None, verbose=True):
+    '''
+    Run a command in subprocess, being able to tail stdout and stderr at the
+    same time.
+
+    Return constructed Popen instance, and a tail function, which helps to
+    interactive with output from stdout and stderr.
+
+    tail is a generator function, and each element is a pair (tag, chunk),
+    where integer tag can be assigned to 0, 1, 2, which gives the meaning
+    of coresponding chunk.
+    0: None for heartbeat, and integer for return code
+    1: None or a chunk of text from stdout
+    2: None or a chunk of text from stderr
+    '''
+    if verbose:
+        logging.info('Command in subprocess: "%s"', ' '.join(cmd))
+
+    sp = subprocess.Popen(
```

Comment:
Whenever I see a subprocess.Popen, I get a little concerned. How will this play with our other signal handling and task killing logic? @bolkedebruin @jlowin